### PR TITLE
ACM-14305: prepare make -C openshift release-chart

### PR DESCRIPTION
**Pull request by marek-veber**
1.  make a chart release -into-> `openschift/chats/cluster-api-operator` directory
```
make -C openshift RELEASE_TAG=v4.18 release-chart
```
2. make a helm chart package:
    * based on the release in `openschift/chats/cluster-api-operator`
    *  -into-> `out/package/cluster-api-operator-4.18.tgz`
```
make -C openshift release-chart-package
```

### CODE REVIEW

File: openshift/Makefile [hunk 1]
```diff
@@ -1,11 +1,33 @@
+RELEASE_TAG   ?= v4.17
+OCP_REGISTRY := registry-proxy.engineering.redhat.com
+IMAGE_NAME   := rh-osbs/openshift-ose-cluster-kube-cluster-api-rhel9-operator
```
Comment (serngawy) [inline openshift/Makefile:3]: should be ose-cluster-kube-cluster-api-rhel9-operator

File: openshift/Makefile [hunk 2]
```diff
@@ -1,11 +1,33 @@
+RELEASE_TAG   ?= v4.17
```
Comment (serngawy) [inline openshift/Makefile:1]: for main branch keep it latest

File: openshift/Makefile [hunk 3]
```diff
@@ -1,11 +1,33 @@
...
+CHART_PACKAGE_DIR := $(RELEASE_DIR)/package
+CHART_VALUES_FILE := $(CHART_DIR)/values.yaml
+CHART_TEMPLATE_FILE := $(CHART_DIR)/templates/deployment.yaml
+
+.PHONY: release-chart
+release-chart:
+	make -C .. STAGING_REGISTRY=$(OCP_REGISTRY) PROD_REGISTRY=$(OCP_REGISTRY) IMAGE_NAME=$(RELEASE_TAG) RELEASE_TAG=$(RELEASE_TAG) release-chart
+	$(YQ) eval '.core      = "xxx"' $(CHART_VALUES_FILE) -i
+	$(YQ) eval '.bootstrap = "yyy"' $(CHART_VALUES_FILE) -i
+	sed -i -e 's;^\([[:space:]]*-[[:space:]]*\)/manager;\1/usr/bin/capi-operator;' $(CHART_TEMPLATE_FILE)
```
Comment (serngawy) [inline openshift/Makefile:32]: should /bin/cluster-api-operator instead of /usr/bin/capi-operator

File: openshift/Makefile (hunk 2 above)
Comment (marek-veber) [reply openshift/Makefile]:
```shell
RELEASE_TAG  ?= $(shell grep FROM Dockerfile.openshift |grep -v ' AS builder' |sed -e 's;.*/;;' -e 's;:.*;;')
```

**Comment by serngawy** [comment]
Would you add the JIRA num into the title , I believe it is related to ACM-14305

File: openshift/Makefile [hunk 4]
```diff
@@ -1,11 +1,33 @@
+RELEASE_TAG   ?= v4.17
+OCP_REGISTRY := registry-proxy.engineering.redhat.com
```
[the remark this answers is not in the document]
Comment (marek-veber) [reply openshift/Makefile]: IMHO: registry.redhat.io/openshift4 ?

File: openshift/Makefile [hunk 5]
```diff
@@ -1,11 +1,40 @@
...
+	$(MAKE) -C .. release-chart RELEASE_TAG=$(RELEASE_TAG)
+	$(MAKE) -C .. set-manifest-image-chart \
+		MANIFEST_IMG=$(OCP_REGISTRY)/$(IMAGE_NAME) MANIFEST_TAG=$(RELEASE_TAG) \
+	        TARGET_RESOURCE="$(CHART_VALUES_FILE)"
+	$(MAKE) -C .. set-manifest-pull-policy-chart \
+	        PULL_POLICY=IfNotPresent \
+	        TARGET_RESOURCE="$(CHART_VALUES_FILE)"
+	# $(YQ) eval '.core      = "xxx"' $(CHART_VALUES_FILE) -i
+	# $(YQ) eval '.bootstrap = "yyy"' $(CHART_VALUES_FILE) -i
+	sed -i -e 's;^\([[:space:]]*-[[:space:]]*\)/manager;\1/bin/cluster-api-operator;' $(CHART_TEMPLATE_FILE)
```
Comment (serngawy) [inline openshift/Makefile:39]: I believe better to save the chart.yaml, values.yaml & template under openshift directory eg; cluster-api-operator/openshift/charts/ with reference to downstream images. So please include a version of the downstream chart with the PR.

File: openshift/Makefile [hunk 6]
```diff
@@ -1,11 +1,40 @@
...
+	$(HELM) package $(CHART_DIR) --app-version=$(HELM_CHART_TAG) --version=$(HELM_CHART_TAG) --destination=$(CHART_PACKAGE_DIR)
```
Comment (serngawy) [inline openshift/Makefile:40]: create another make tag for packaging the chart eg; release-chart-package. The tar file could be stored under cluster-api-operator/out/

**Review by serngawy — CHANGES_REQUESTED** [review]

File: openshift/charts/cluster-api-operator/templates/ocp-capa-provider.yaml [hunk 7]
```diff
@@ -0,0 +1,69 @@
+apiVersion: v1
```
Comment (serngawy) [inline openshift/charts/cluster-api-operator/templates/ocp-capa-provider.yaml:1]: please remove this file from this PR , will include it in another PR with jira ACM-14308

File: openshift/charts/cluster-api-operator/templates/ocp-capi-provider.yaml [hunk 8]
```diff
@@ -0,0 +1,45 @@
+apiVersion: v1
```
Comment (serngawy) [inline openshift/charts/cluster-api-operator/templates/ocp-capi-provider.yaml:1]: same as ocp-capa-provider.yaml file

File: openshift/Makefile [hunk 9]
```diff
@@ -1,11 +1,52 @@
...
+	        TARGET_RESOURCE="$(CHART_VALUES_FILE)"
+	$(MAKE) -C .. set-manifest-pull-policy-chart \
+	        PULL_POLICY=IfNotPresent \
+	        TARGET_RESOURCE="$(CHART_VALUES_FILE)"
+	sed -i -e 's;^\([[:space:]]*-[[:space:]]*\)/manager;\1/bin/cluster-api-operator;' $(CHART_TEMPLATE_FILE)
+	mkdir -p $(shell dirname $(OC_CHART_DIR))
+	cp -fr $(CHART_DIR)/. $(OC_CHART_DIR)
+	$(YQ) eval '.version      = $(HELM_CHART_TAG)' $(OC_CHART_YAML) -i
+	$(YQ) eval '.appVersion = "$(HELM_CHART_TAG)"' $(OC_CHART_YAML) -i
+	$(YQ) eval '.AWS_B64ENCODED_CREDENTIALS = "ZmFrZQo="' $(CHART_VALUES_FILE) -i # fake base64 string
```
Comment (serngawy) [inline openshift/Makefile:44]: no need to set this now

**Review by serngawy — CHANGES_REQUESTED** [review]

File: openshift/Makefile (hunk 5 above)
Comment (marek-veber) [reply openshift/Makefile:39]: done

**Comment by serngawy** [comment]
@marek-veber please squash the commits

**Comment by serngawy** [comment]
As we discussed under the template directory we only need the deployment.yaml, operator-component.yaml and _helpers.tpl other template files no need for them.

**Pull request merged**